pkg/koordlet/statesinformer/states_device.go

```diff
@@ -0,0 +1,232 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package statesinformer
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+	schedulingv1alpha1 "github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/metriccache"
+)
+
+func generateQueryParam() *metriccache.QueryParam {
+	end := time.Now()
+	start := end.Add(-time.Duration(60) * time.Second)
+	return &metriccache.QueryParam{
+		Aggregate: metriccache.AggregationTypeLast,
+		Start:     &start,
+		End:       &end,
+	}
+}
+
+func (s *statesInformer) reportDevice() {
+	s.nodeRWMutex.RLock()
+	copyNode := s.node.DeepCopy()
+	s.nodeRWMutex.RUnlock()
+	gpuDevices := s.buildGPUDevice()
+	blocker := true
+	device := &schedulingv1alpha1.Device{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: copyNode.Name,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         "v1",
+					Kind:               "Node",
+					Name:               copyNode.Name,
+					UID:                copyNode.UID,
+					Controller:         &blocker,
+					BlockOwnerDeletion: &blocker,
+				},
+			},
+		},
+		Spec: schedulingv1alpha1.DeviceSpec{
+			Devices: gpuDevices,
+		},
+	}
+	// TODO: compare diff before update
```

Comment:
please create an issue and add the link here, maybe also include other related crds like `NodeTopo`?